### PR TITLE
:new: Sticker component

### LINKED ISSUE #440: ODS > Stickers
## Requirement

**For** a Developer 
**Who** is creating an Orange Website
**The** Sticker Integration
**is an** automated component
**that** proposes the web site's Orange Sticker component to promote content
**unlike** the default bootstrap framework
**Boosted** will provide this ready to use component compatible with the Orange Design and the accessibility requirements

## Description

Started as a POC using SVG [in v5 Design Guidelines](https://v5-dev--boosted.netlify.app/docs/5.0/guidelines/stickers/) – but that need to be tested for accessibility and browser support, obviously.

If we stick to SVG:

1. use `<title id="">`
2. use `aria-labelledby` on `<svg>`
3. maybe use `role="img"`?
4. Regarding accessibility, maybe have a look at a11y-guildeines: https://a11y-guidelines.orange.com/fr/web/exemples-de-composants/vocalisation-des-prix/

## References

* **Orange Design System**: [Components > Stickers](https://system.design.orange.com/0c1af118d/p/723617-stickers/b/87f656)
  * Service Offer On One Line — WEB-PRM-STK-001
  * Service Offer Icon — WEB-PRM-STK-002
  * Service Offer On Two Lines — WEB-PRM-STK-003
  * Service Offer On Two Lines With Icon — WEB-PRM-STK-004
  * With Price — WEB-PRM-STK-005
  * With Old And New Price — WEB-PRM-STK-006
  * Bonus Minutes — WEB-PRM-STK-007
  * Bonus Offer — WEB-PRM-STK-008
  * Additional Minutes — WEB-PRM-STK-009
  * Additional Offer — WEB-PRM-STK-010

### PR DESCRIPTION
**Pull request by julien-deramond**
Closes #440 

Agreed with the designers that we can simply implement a container without placing elements with pixel perfect.

### Live Previews
* [Documentation](https://deploy-preview-1023--boosted.netlify.app/docs/5.1/components/sticker/)
* [Examples](https://deploy-preview-1023--boosted.netlify.app/docs/5.1/examples/stickers/)

### Development

- [x] Should match specs (eg. either the Web UI Kit or any pattern from the WAI — or both…)
- [x] Docs added:
  - including the "Sass" part using `scss-docs` shortcode
  - in /about/overview/#custom-components if it is a new Orange custom component
  - in /getting-started/introduction/#components if it is a new Orange custom component that requires JavaScript (and Popper)
  - in /customize/overview#csps-and-embedded-svgs if it is a new Orange custom component that includes embedded SVGs in our CSS
  - in /forms/validation/?#supported-elements if it is a new Orange custom component that is a form control
  - in /forms/overview/ if it is a new Orange custom component that is a form control
- [x] Check (and fix) RTL version
- [x] Run linters
- [x] Run compilers
- [x] Manually run BrowserStack test
- [x] Manually run Percy test
- [x] Cross-browser test:
  - Firefox ESR
  - Latest Edge, Chrome, Firefox, Safari
  - iOS Safari
  - Chrome & Firefox on Android

### Reviews

- [x] Code review
- [x] A11y review
- [x] Design review
  - [x] Update the thumbnails after the design review

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -1838,4 +1838,15 @@ $step-link-marker-lg:    counter($stepped-process-counter) ".\A0" !default;
...
 // scss-docs-end stepped-process
+
+//// Sticker
+// scss-docs-start sticker
+$sticker-size-sm:                       $spacer * 7 !default;
+$sticker-size-md:                       $spacer * 9 !default;
+$sticker-size-lg:                       $spacer * 14 !default;
+
+$sticker-content-max-width-sm:          $sticker-size-sm * .7 !default;
+$sticker-content-max-width-md:          8.1875rem !default; // Should be $sticker-size-md * .7;
```
Comment (julien-deramond) [inline scss/_variables.scss:1849]: - [ ] Check with the designers this behavior.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+    <div class="sticker sticker-sm">
+      <svg width="35" height="35" aria-hidden="true" focusable="false">
+        <use xlink:href="/docs/{{< param docs_version >}}/assets/img/boosted-sprite.svg#delivery"></use>
+      </svg>
+      <span class="h4 mb-2">Free delivery</span>
+    </div>
+
+    <div class="sticker sticker-sm">
+      <span class="h2 mb-0">39,99€</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:33]: `.mt-1` to better fit specs.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    <div class="sticker">
+      <svg width="40" height="40" aria-hidden="true" focusable="false">
+        <use xlink:href="/docs/{{< param docs_version >}}/assets/img/boosted-sprite.svg#delivery"></use>
+      </svg>
+      <span class="h3 mb-3">Free delivery</span>
+    </div>
+
+    <div class="sticker">
+      <span class="display-3 mb-1">39,99€</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:92]: `mt-1` to better fit specs.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      <span class="h3 mb-0 text-decoration-line-through" aria-hidden="true">69,99€</span>
+      <span class="display-3 mb-1" aria-hidden="true">39,99€</span>
+      <span class="mb-1" aria-hidden="true">Per month</span>
+      <span class="visually-hidden">39,99€ per month instead of 69,99€</span>
+    </div>
+
+    <div class="sticker">
+      <span class="mb-1">Bonus</span>
+      <span class="display-3 mb-1">90mins</span>
+      <span class="mb-1">Per month</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:106]: I would have done `.mb-1`, `.mb-0` and `.mb-2` on these `<span>`.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    </div>
+
+    <div class="sticker">
+      <span class="mb-1">Bonus</span>
+      <span class="display-3 mb-1">250%</span>
+      <span class="mb-0">With Orange Money</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:112]: Same as above.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+    <div class="sticker sticker-lg">
+      <svg width="62" height="62" aria-hidden="true" focusable="false">
+        <use xlink:href="/docs/{{< param docs_version >}}/assets/img/boosted-sprite.svg#delivery"></use>
+      </svg>
+      <span class="display-3 mb-4">Free delivery</span>
+    </div>
+
+    <div class="sticker sticker-lg">
+      <span class="display-1 mb-1">39,99€</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:151]: `.mt-2` to better fit specs

File: site/content/docs/5.1/examples/stickers/index.html [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+      <span class="h3 mb-0">Per month</span>
+    </div>
+
+    <div class="sticker sticker-lg">
+      <span class="display-4 mb-0 text-decoration-line-through" aria-hidden="true">69,99€</span>
+      <span class="display-1 mb-1" aria-hidden="true">39,99€</span>
+      <span class="h3 mb-1" aria-hidden="true">Per month</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:158]: I would have done `.mb-1`, `.mb-1` and `.mb-2` on these `<span>`.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+      <span class="visually-hidden">39,99€ per month instead of 69,99€</span>
+    </div>
+
+    <div class="sticker sticker-lg">
+      <span class="h3 mb-1">Bonus</span>
+      <span class="display-1 mb-1">90mins</span>
+      <span class="h3 mb-1">Per month</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:165]: I would have done `.mb-2`, `.mb-1` and `.mb-3` on these `<span>`.

File: site/content/docs/5.1/examples/stickers/index.html [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    </div>
+
+    <div class="sticker sticker-lg">
+      <span class="h3 mb-1">Bonus</span>
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:169]: `.mb-2` to better fit specs

File: site/content/docs/5.1/examples/stickers/index.html [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+      <span class="h3 mb-1">With Orange Money</span>
+    </div>
+
+    <div class="sticker sticker-lg">
+      <span class="display-1 mb-0">+</span>
+      <span class="display-1 mb-1">90mins</span>
+      <span class="h3 mb-4">Per month</span>
+    </div>
+
+    <div class="sticker sticker-lg">
```
Comment (louismaximepiton) [inline site/content/docs/5.1/examples/stickers/index.html:180]: `.pb-3` to better fit specs

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
Is the responsive size of font-size wanted ? -> Many spacing issues in small viewports
Maybe put a line to tell that utilities works on `.sticker` too ?

File: scss/_variables.scss (hunk 1 above)
Comment (julien-deramond) [reply scss/_variables.scss]: FI, I changed the value from .7 to a more precise one using √2 / 2 -> .7071067812.

For the 3 sizes, the calculated width/height are:
```
scss/_variables.scss:1853 Debug: 6.1871843355rem
scss/_variables.scss:1854 Debug: 7.9549512885rem
scss/_variables.scss:1855 Debug: 12.374368671rem
```

If you draw the inner square:

Capture d’écran 2022-01-10 à 08 08 26

**Review by louismaximepiton — APPROVED** [review]
Sounds good to me.

**Comment by julien-deramond** [comment]
A11y feedback from @Aniort: include the sticker's content within a `<p>` for semantics.

**Comment by julien-deramond** [comment]
Last thing to do in this PR: explain that our example in not a responsive example and is just a simple example with utility classes that works for desktop.

**Pull request merged**